### ISSUE TITLE
Headless compilation errors stick into memory somewhere

### ISSUE THREAD

**Issue by romainreuillon** [gama-platform/gama#2311]
In OpenMOLE when we get a compilation error, for instance:

```
javax.script.ScriptException: org.openmole.core.exception.UserBadDataError: Gaml compilation errors (some errors may be caused by OpenGL function calls, in this case either separate the graphical aspect in another gaml file or set "failOnGamlError = false" on the GAMATask):
  INFO: This definition of color supersedes the one in generic_species
  INFO: This definition of max_energy supersedes the one in generic_species
  INFO: This definition of max_transfert supersedes the one in generic_species
  INFO: This definition of energy_consum supersedes the one in generic_species
  INFO: This definition of color supersedes the one in generic_species
  INFO: This definition of max_energy supersedes the one in generic_species
  INFO: This definition of energy_consum supersedes the one in generic_species
  INFO: This definition of color supersedes the one in skill grid
  INFO: This definition of color supersedes the one in generic_species
  INFO: This definition of max_energy supersedes the one in generic_species
  INFO: This definition of max_transfert supersedes the one in generic_species
  INFO: This definition of energy_consum supersedes the one in generic_species
  INFO: This definition of color supersedes the one in generic_species
  INFO: This definition of max_energy supersedes the one in generic_species
  INFO: This definition of energy_consum supersedes the one in generic_species
  ERROR: Parameter 'Test List'  must have an initial value
	at org.openmole.core.console.OMScripted$WrappedRequest.$anonfun$eval$1(OMScripted.scala:189)
	at org.openmole.core.console.OMScripted.withScriptContext(OMScripted.scala:87)
	at org.openmole.core.console.OMScripted$WrappedRequest.eval(OMScripted.scala:186)
	at javax.script.CompiledScript.eval(CompiledScript.java:106)
	at org.openmole.core.console.ScalaREPL.$anonfun$compile$1(ScalaREPL.scala:287)
	at org.openmole.core.project.Compiled.eval(Project.scala:102)
	at org.openmole.gui.server.core.ApiImpl.$anonfun$runScript$3(ApiImpl.scala:337)
	at org.openmole.core.outputmanager.OutputManager$.withStreamOutputs(OutputManager.scala:115)
	at org.openmole.gui.server.core.ApiImpl.$anonfun$runScript$2(ApiImpl.scala:337)
	at org.openmole.gui.server.core.ApiImpl.catchAll$1(ApiImpl.scala:330)
	at org.openmole.gui.server.core.ApiImpl.$anonfun$runScript$1(ApiImpl.scala:337)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.openmole.core.exception.UserBadDataError: Gaml compilation errors (some errors may be caused by OpenGL function calls, in this case either separate the graphical aspect in another gaml file or set "failOnGamlError = false" on the GAMATask):
  INFO: This definition of color supersedes the one in generic_species
  INFO: This definition of max_energy supersedes the one in generic_species
  INFO: This definition of max_transfert supersedes the one in generic_species
  INFO: This definition of energy_consum supersedes the one in generic_species
  INFO: This definition of color supersedes the one in generic_species
  INFO: This definition of max_energy supersedes the one in generic_species
  INFO: This definition of energy_consum supersedes the one in generic_species
  INFO: This definition of color supersedes the one in skill grid
  INFO: This definition of color supersedes the one in generic_species
  INFO: This definition of max_energy supersedes the one in generic_species
  INFO: This definition of max_transfert supersedes the one in generic_species
  INFO: This definition of energy_consum supersedes the one in generic_species
  INFO: This definition of color supersedes the one in generic_species
  INFO: This definition of max_energy supersedes the one in generic_species
  INFO: This definition of energy_consum supersedes the one in generic_species
  ERROR: Parameter 'Test List'  must have an initial value
	at org.openmole.plugin.task.gama.GamaTask$.extensionPlugins(GamaTask.scala:111)
	at org.openmole.plugin.task.gama.GamaTask$.apply(GamaTask.scala:62)
	at $line16.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$$$a5cddfc4633c5dd8aa603ddc4f9aad5$$$$w$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$.runOMSScript(<console>:208)
	at $line16.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$$$a5cddfc4633c5dd8aa603ddc4f9aad5$$$$w$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$.<init>(<console>:221)
	at $line16.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$$$a5cddfc4633c5dd8aa603ddc4f9aad5$$$$w$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw$.<clinit>(<console>)
	at $line16.$eval$.$result$lzycompute(<console>:5)
	at $line16.$eval$.$result(<console>:5)
	at $line16.$eval.$result(<console>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.call(IMain.scala:733)
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.callEither(IMain.scala:737)
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.evalEither(IMain.scala:752)
	at org.openmole.core.console.OMScripted$WrappedRequest.$anonfun$evalEither$3(OMScripted.scala:168)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at scala.Console$.withIn(Console.scala:226)
	at org.openmole.core.console.OMScripted$WrappedRequest.$anonfun$evalEither$2(OMScripted.scala:168)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at scala.Console$.withErr(Console.scala:192)
	at org.openmole.core.console.OMScripted$WrappedRequest.$anonfun$evalEither$1(OMScripted.scala:167)
	at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
	at scala.Console$.withOut(Console.scala:163)
	at org.openmole.core.console.OMScripted$WrappedRequest.evalEither(OMScripted.scala:166)
	at org.openmole.core.console.OMScripted$WrappedRequest.$anonfun$eval$1(OMScripted.scala:187)
	... 14 more
Caused by: msi.gama.headless.core.GamaHeadlessException: Model cannot be compiled. See list of attached errors
	at msi.gama.headless.core.HeadlessSimulationLoader.loadModel(HeadlessSimulationLoader.java:131)
	at msi.gama.headless.openmole.MoleSimulationLoader.loadModel(MoleSimulationLoader.java:50)
	at org.openmole.plugin.task.gama.GamaTask$.$anonfun$extensionPlugins$1(GamaTask.scala:109)
	at org.openmole.plugin.task.gama.GamaTask$.org$openmole$plugin$task$gama$GamaTask$$withDisposable(GamaTask.scala:82)
	at org.openmole.plugin.task.gama.GamaTask$.extensionPlugins(GamaTask.scala:109)
	... 39 more
```

the error stick somewhere into memory and arise in any case, even if we fix the error in the gaml file and compile it again. I know this is a bit of a fuzzy report but do you have any idea why it could append or how I can get more info about to diagnose it?

Cheers,
Romain

**Comment by AlexisDrogoul** [comment]
It depends on how you reinitiate GAMA after "fixing" the error. Do you relaunch everything from the beginning ?

**Comment by romainreuillon** [comment]
That's the intension, I call loadModel, here is the code:
https://github.com/openmole/gama-plugin/blob/7-dev/org.openmole.plugin.task.gama/src/main/scala/org/openmole/plugin/task/gama/GamaTask.scala#L155

Shall I call something to clean the errors, or shall I close something?

**Comment by AlexisDrogoul** [comment]
Now tracked under #2453

**Comment by gnoubi** [comment]
compilation errors is now catch and add to log file

**Comment by hqnghi88** [comment]
Dear,
I think i fall into this state "the error stick somewhere into memory and arise in any case, even if we fix the error in the gaml file and compile it again." , so @romainreuillon did you success to fix this ? any comment would be nice.
Thanks

**Comment by AlexisDrogoul** [comment]
Probably solved and fixed by the fix to #2708 -- unless I'm mistaking the issues ?
